### PR TITLE
Introducing anelasticity corrections for P wave speeds. 

### PR DESCRIPTION
**Pull request by sghelichkhani**
# Pull Request: Add P-Anelastic Correction and Update Plotting

## Summary
This PR introduces a P-anelastic correction model by extending the `anelasticity` framework. Key updates:
1. Added **Q_mu** and **Q_kappa** to calculate **Q_s_wave** and **Q_p_wave** with equations and references.
2. Updated the example script to plot P-wave model changes.

---

### 1. Anelastic Correction
- **New Parameters:** `Q_mu` (shear attenuation) and `Q_kappa` (bulk attenuation).
- **Equations:**
  - \( Q_s = Q_\mu \)
  - \( Q_p = \frac{(4/3)\mu + \kappa}{(4/3)\mu / Q_\mu + \kappa / Q_\kappa} \)
- Integrated Q-value interpolation in `compute_swave_speed` and `compute_pwave_speed`.

### 2. Example Updates
- Added plots for P-wave velocity changes due to anelastic corrections.
- Enhanced plot labels and legends for clarity.

### CODE REVIEW

File: examples/01_anelasticity_models.py [hunk 1]
```diff
@@ -6,19 +6,23 @@
...
 
 def build_solidus():
     # Defining the solidus curve for manlte
+    # Andrault et al 2011
     andrault_solidus = gdrift.SolidusProfileFromFile(
         model_name="1d_solidus_Andrault_et_al_2011_EPSL",
         description="Andrault et al 2011 EPSL")
 
-    # Defining parameters for Cammarano style anelasticity model
+    # Hirsch solidus
```
Comment (drhodrid) [inline examples/01_anelasticity_models.py:22]:
```suggestion
    # Hirschman solidus
```

File: examples/01_anelasticity_models.py [hunk 2]
```diff
@@ -6,19 +6,23 @@
 # In this tutorial we show how with the given functionalities
 # we can apply anelastic correction to an existing thermodynamic table
 
-
 # For our anelastic model in this example we use the study by Cammerano et al.
+# The anelastic parameterisation is dependent on a solidus curve of the mantle.
+# To this end we use the two datasets by Andrault et al 2011 EPSL
+# And a solidus by a study by Hirschmann to build a combined solidus curve.
```
Comment (drhodrid) [inline examples/01_anelasticity_models.py:12]: Add journal and year to Hirschmann paper.

File: examples/01_anelasticity_models.py [hunk 3]
```diff
@@ -27,9 +31,11 @@ def build_solidus():
         my_depths.extend(dpths)
         my_solidus.extend(solidus_model.at_depth(dpths))
 
+    # Avoding unnecessary extrapolation by setting the solidus temperature at maximum depth
     my_depths.extend([3000e3])
     my_solidus.extend([solidus_model.at_depth(dpths[-1])])
 
+    # building the solidu profile that was originally used by Ghelichkhan et al in their study
```
Comment (drhodrid) [inline examples/01_anelasticity_models.py:38]:
```suggestion
    # building the solidus profile that was originally used by Ghelichkhan et al. (2021)
```

File: examples/01_anelasticity_models.py [hunk 4]
```diff
@@ -38,27 +44,42 @@ def build_solidus():
     return ghelichkhan_et_al
 
 
+# Now the anelasticiy model by Cammarano et al can be constructed.
+# Here we use B g a for making Q_\mu. and we provide a rough Q_\kappa
```
Comment (drhodrid) [inline examples/01_anelasticity_models.py:48]:
```suggestion
# Here we use B g a for making Q_\mu, and we provide a rough Q_\kappa
```

File: examples/01_anelasticity_models.py [hunk 5]
```diff
@@ -38,27 +44,42 @@ def build_solidus():
...
+# Provided by Goes et al.
```
Comment (drhodrid) [inline examples/01_anelasticity_models.py:49]:
```suggestion
# From Goes et al, set to a constant value of 1000 in the upper and 10000 in the lower mantle.
```

**Comment by drhodrid** [comment]
This looks really good to me. I made some minor comments on the code. In the example, I think it's important to emphasize that you are using model Q1 of Cammarano. 

One other thing that would be nice moving forward, but is not necessary for this pull request -- the ability to pick a set of parameter values, without having to put them in individually (e.g. models Q1, Q4, or Q6 from Cammarano / Goes). 

I will ask @HainingChang to test this further this week and report back, but to me it looks good to go.

**Review by drhodrid — APPROVED** [review]

**Comment by drhodrid** [comment]
The recent edits to include the Cammarano parameter values (I checked and these are all correct) are nice and make this much more user friendly. As @tom-new has tested this, and a few minor bugs have been fixed, I'd say this is once again good to go.

**Comment by tom-new** [comment]
@drhodrid I think it would be even nicer for the end-user if we implement the Cammarano parameter values as part of the `CammaranoAnelasticityModel` class itself, as an additional constructor which can just be passed the name of the Q-model as a string. This would save the user from having to construct the parameters themselves. I can submit a PR for this as well soon™, since I'm doing this anyway for the Goes parameters/equations in my attempt to comare the two papers.

**Comment by sghelichkhani** [comment]
Thanks @drhodrid and @tom-new. I'll merge now and then continue to include Tom's suggestions. There will be a new PR soon on linearisation.

**Pull request merged**